src/llscan.cc

```diff
@@ -768,16 +769,45 @@ void LLScan::ScanMemoryRanges(FindJSObjectsVisitor& v) {
     /* Brute force search - query every address - but allow the visitor code to
      * say how far to move on so we don't read every byte.
      */
-    for (auto searchAddress = address; searchAddress < address + len;) {
-      uint32_t increment =
-          v.Visit(searchAddress, (address + len) - searchAddress);
+
+    SBError sberr;
+    uint64_t address_end = address + len;
+
+    // Load data in blocks to speed up whole process
+    for (auto searchAddress = address; searchAddress < address_end;
+         searchAddress += block_size) {
+      size_t loaded = std::min(address_end - searchAddress, block_size);
+      process_.ReadMemory(searchAddress, block, loaded, sberr);
+      if (sberr.Fail()) {
+        // TODO(indutny): add error information
+        break;
+      }
+
+      uint32_t increment = 1;
+      for (size_t j = 0; j + addr_size <= loaded;) {
+        uint64_t value;
+
+        if (addr_size == 4)
+          value = *reinterpret_cast<uint32_t*>(&block[j]);
+        else if (addr_size == 8)
+          value = *reinterpret_cast<uint64_t*>(&block[j]);
+        else
+          break;
+
+        increment = v.Visit(j + searchAddress, value);
+        if (increment == 0) break;
+
+        j += static_cast<size_t>(increment);
+      }
+
       if (increment == 0) {
```

Comment:
I believe @hhellyer meant end of iteration with this. It doesn't seem to be used right now, though.